### PR TITLE
Add Syslog persistence

### PR DESCRIPTION
**Pull request by sysadmin4j**
Adding a SyslogDao based on the **Jenkins Syslog Logger Plugin**.
https://wiki.jenkins-ci.org/display/JENKINS/Syslog+Logger+Plugin

The main goal was to support the CEE format which made logs parsing way easier on the Rsyslog side.  

Hope this can be usefull for someone else!

Ref:
http://www.rsyslog.com/json-elasticsearch/
https://issues.jenkins-ci.org/browse/JENKINS-28040

### CODE REVIEW

**Comment by R-Gerard** [comment]
Cool, thanks for submitting this. There's an open discussion about where this functionality belongs (the Logstash plugin vs. the Syslog plugin): https://issues.jenkins-ci.org/browse/JENKINS-28040
Feel free to chime in on the discussion if you want.

The Travis CI build failed on JDK 1.6. It looks like the Syslog client is compiled on JDK 1.7: 
https://travis-ci.org/jenkinsci/logstash-plugin/jobs/64303150#L120
(...I'm not sure why Travis CI didn't post a commit status for the PR like it normally does.)

I've made it a point to keep backwards compatibility with Java 6 because that's all that's required to run the Jenkins CI server, but maybe it's time to cut the cord on 6? (Since projects released by Cloudbees are starting to use 7.)

File: src/test/java/jenkins/plugins/logstash/persistence/SyslogDaoTest.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import org.junit.Test;
+
+public class SyslogDaoTest {
+  SyslogDao dao;
+  
+  @Test
+  public void send() throws Exception {
+    String data = "{ 'junit': 'SyslogDaoTest' }";
+    dao = createDao("localhost", 514, null, null, null);
+    dao.push(data);
```
Comment (R-Gerard) [inline src/test/java/jenkins/plugins/logstash/persistence/SyslogDaoTest.java]: This isn't a very useful test. You could comment-out all of the code in the push method and this test would still pass.

I'd recommend mocking out the messageSender object and at a minimum verify that the right payload was sent. Take a look at the other DAOs and their tests for examples.

**Comment by sysadmin4j** [comment]
Thanks for your feedback!

For the backwards compatibility with Java 6, let me know if it's required?
I can look to use Syslog4j instead or something similar.

**Comment by sysadmin4j** [comment]
For the backwards compatibility with Java 6, let me know if it's required?
I can look to use Syslog4j instead or something similar.

**Comment by R-Gerard** [comment]
Sorry for not reviewing this sooner... It looks fine to me. After the discussion on the Jira issue I think we should move forward and cut the JDK 6 support so we can use the same library as the Syslog plugin.

Do you mind squashing your changes into a single commit?

**Comment by sysadmin4j** [comment]
Thanks @R-Gerard for adding it to the new 1.1.0 version.

**Pull request merged**